Lib/idlelib/idle_test/test_debugger.py

```diff
@@ -23,7 +34,371 @@ def test_init(self):
         debugger.NamespaceViewer(self.root, 'Test')
 
 
-# Other classes are Idb, Debugger, and StackViewer.
+class MockFrameType:
+    """Reflection of the types.FrameType."""
+    f_back = None
+    f_builtins = None
+    f_code = None
+    f_globals = None
+    f_lasti = None
+    f_lineno = None
+    f_locals = None
+    f_restricted = 0
+    f_trace = None
+
+    def __init__(self, code, lineno):
+        self.f_code = code
+        self.f_lineno = lineno
+
+
+class IdbTest(unittest.TestCase):
+
+    def setUp(self):
+        self.gui = mock.Mock()
+        self.gui.interaction = mock.Mock()
+        self.idb = debugger.Idb(self.gui)
+
```

Comment:
When you create instances of the class for debugging (such as `self.idb`), you also need to define a `tearDown()` function to `close()`/`destroy()`/delete (`del`) (whatever is appropriate for your object) to cleanup what's been created.  At a minimum, this helps prevent reference leaks (refleaks).   When the instances are created in `setUp()`, then a new one is created for every test in this test class and thus needs to be cleaned up after each individual test is run. 

Just looking over this quickly, you may be able to have the mock created at a `setUpClass` level and then call `reset_mock` during `setUp` (or `tearDown`) to clear the mock before the next test.